### PR TITLE
[googlemaps] improve map types & docs

### PR DESCRIPTION
**Pull request by demensky**
-   [`google.maps.MVCObject`](https://developers.google.com/maps/documentation/javascript/reference/event#MVCObject).
    -   Added `MVCEventHandler` for convenient writing handlers.
-   [`google.maps.Map`](https://developers.google.com/maps/documentation/javascript/reference/map).
    -   Add many JSDoc! 😁
    -   Make `mapDiv` (`Map#getDiv`) generic.
    -   Fix `mapDiv` (not nullable).
    -   Add strict `addListener` handler types.
    -   Add the missing [`getClickableIcons`](https://developers.google.com/maps/documentation/javascript/reference/map#Map.getClickableIcons) method.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developers.google.com/maps/documentation/javascript/reference/map#Map.setStreetView
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by oBusk** [comment]
@demensky The changes to `load-google-maps-api` tests looks fine. However, I don't see why the TypeScript version had to be bumped to 3.0 seeing how there was no changes to the typings. I understand that maybe the new google maps typings could require a new TS version, but seems unecessary to version bump load-google-maps-api when there are now changes.

**Comment by demensky** [comment]
@oBusk 

Without bumped TypeScript:

```sh
npm test
```

screen

**Comment by oBusk** [comment]
@demensky ok! Fair enough. `load-google-maps-api` LGTM, but I'll not approve review so that someone of  @cgwrench @Silver-Connection @nertzy @xaolas @mrmcnerd @martincostello @svenkreiss @bolatovumar @gauthierm @captain-igloo @life777 @simonhaenisch has a look on `googlemaps`.

**Comment by oBusk** [comment]
Well that's not very helpful @typescript-bot . I hope maintainer looks before merging.

**Comment by demensky** [comment]
@RyanCavanaugh

**Review by simonhaenisch — COMMENTED** [review]
LGTM but I didn't check all the links

File: types/load-google-maps-api/load-google-maps-api-tests.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import loadGoogleMapsApi = require('load-google-maps-api');
 
 // Example from README.md https://github.com/yuanqing/load-google-maps-api#usage
 loadGoogleMapsApi().then((googleMaps) => {
-    new googleMaps.Map(document.querySelector('.map'), {
```
Comment (orta) [inline types/load-google-maps-api/load-google-maps-api-tests.ts]: Is this a breaking change? If so you should probably bump the semver (plus the TS min version change makes for good reasoning IMO)

Comment (demensky) [reply types/load-google-maps-api/load-google-maps-api-tests.ts]: In versions [3.39] (current), [3.38], and [3.37], Map always requires an element.

[3.39]: https://developers.google.com/maps/documentation/javascript/reference/map#Map.constructor
[3.38]: https://developers.google.com/maps/documentation/javascript/reference/3.38/map#Map.constructor
[3.37]: https://developers.google.com/maps/documentation/javascript/reference/3.37/map#Map.constructor

**Comment by orta** [comment]
OK cool, then I'd like one more +1 and it's good to go

**Comment by andrewbranch** [comment]
I see two owner LGTMs even though no one wanted to push approve, and nobody else came after a week. Works for me.

**Pull request merged**